### PR TITLE
Include variable scope clause in deprecated Cypher query

### PR DESCRIPTION
**Pull request by willtai**
# Summary
The previous Cypher query would include a warning that the Cypher query will be deprecated. This PR update the query to have a variable scope clause. See https://neo4j.com/docs/cypher-manual/current/subqueries/call-subquery/#variable-scope-clause

This PR also adds a unit test and was manually tested.

### CODE REVIEW

**Review by alexthomas93 — APPROVED** [review]
LGTM

File: libs/neo4j/langchain_neo4j/vectorstores/neo4j_vector.py [hunk 1]
```diff
@@ -93,7 +93,7 @@ def _get_search_index_query(
                 "YIELD node, score "
             ),
             SearchType.HYBRID: (
-                "CALL { "
+                "CALL () { "
```
Comment (stellasia) [inline libs/neo4j/langchain_neo4j/vectorstores/neo4j_vector.py:96]: Are we supporting only neo4j >= 5.23 ?

Comment (willtai) [reply libs/neo4j/langchain_neo4j/vectorstores/neo4j_vector.py:96]: Good point, I should include the old queries for older versions

Comment (willtai) [reply libs/neo4j/langchain_neo4j/vectorstores/neo4j_vector.py]: https://github.com/langchain-ai/langchain-neo4j/pull/6/commits/ed09e73c1b955a3ea3be47fe90438ad0185abf70

Comment (alexthomas93) [reply libs/neo4j/langchain_neo4j/vectorstores/neo4j_vector.py]: I created another ticket to add support for Neo4j < 5.23 across the package. Feel free to add in support here but ideally we should think about exactly which versions we want to support and come up with a strategy that works across the package.

**Pull request merged**